### PR TITLE
[MB-1342] Unique constraint violation fix

### PR DESCRIPTION
**Pull request by pumudu88**
This PR will fix following jira.
https://wso2.org/jira/browse/MB-1342

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 1]
```diff
@@ -1392,9 +1394,18 @@ private void createNewQueue(final Connection connection, final String destinatio
...
         } catch (SQLException e) {
-            log.error("Error occurred while inserting destination queue [" + destinationQueueName +
-                    "] to database ");
-            throw e;
+            AndesException andesException =
+                    rdbmsStoreUtils.convertSQLException("Error occurred while creating queue", e);
+
+            if(andesException instanceof AndesDataIntegrityViolationException) {
+                // This exception occurred because some other node has created the queue in parallel.
+                // Therefore no need to create the queue. It's already created.
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: may be we can print a WARN here?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java [hunk 2]
```diff
@@ -1392,9 +1394,18 @@ private void createNewQueue(final Connection connection, final String destinatio
...
+                // Nothing need to be done if this exception occur.
+            } else {
+                log.error("Error occurred while inserting destination queue [" + destinationQueueName +
+                          "] to database ");
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: should we be logging with the exception?

**Comment by ramith** [comment]
can we merge two commits.?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java (hunk 1 above)
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java]: +1

**Pull request merged**